lib/java/logging-service-telemetry/src/main/java/org/enso/logging/service/telemetry/LogJobsProcessor.java

```diff
@@ -220,6 +207,18 @@ private void sendLogRequest(HttpRequest request, int retryCount) throws RequestF
     }
   }
 
+  private AuthenticationData fetchNewAccessToken() {
+    for (int i = 0; i < MAX_RETRIES; i++) {
+      var refreshedAuthData = tokenRefresher.fetchNewAccessToken();
+      if (refreshedAuthData != null) {
+        return refreshedAuthData;
+      } else {
+        LOGGER.warn("Failed to refresh token. Retrying... (attempt {}/{})", i + 1, MAX_RETRIES);
```

Comment:
I would expect to see _some_ exponential backoff here.

Reply:
That was my original intent. But I have no idea how to simply implement that with JDK. Any suggestions?